core/src/main/java/org/elasticsearch/action/bulk/BulkProcessor.java

```diff
@@ -161,6 +163,14 @@ public Builder setThreadPool(ThreadPool threadPool) {
             return this;
         }
 
+        /**
+         * Sets an optional BiFunction to schedule flush actions and request retries
+         */
+        public Builder setScheduleFunction(BiFunction<TimeValue, Runnable, ScheduledFuture<?>> scheduleFunction) {
+            this.scheduleFunction = scheduleFunction;
```

Comment:
It's not being used right now. I had added this as I was exploring passing a lambda to the builder. I'll provide a more in-depth comment about where this PR is currently a little later today.